packages/strapi-plugin-i18n/config/functions/migrations/field.js

```diff
@@ -0,0 +1,171 @@
+'use strict';
+
+const { difference, pick, orderBy, prop, intersection } = require('lodash/fp');
+const { getService } = require('../../../utils');
+
+const BATCH_SIZE = 1000;
+
+const shouldBeProceed = processedLocaleCodes => entry =>
+  entry.localizations.length > 1 &&
+  intersection(entry.localizations.map(prop('locale')), processedLocaleCodes).length === 0;
+
+const migrateForBookshelf = async ({ ORM, model, attributesToMigrate, locales }) => {
+  // Create tmp table with all updates to make (faster than make updates one by one)
+  const TMP_TABLE_NAME = '__tmp__i18n_field_migration';
+  const columnsToCopy = ['id', ...attributesToMigrate];
+  await ORM.knex.schema.dropTableIfExists(TMP_TABLE_NAME);
+  await ORM.knex.raw(`CREATE TABLE ?? AS ??`, [
+    TMP_TABLE_NAME,
+    ORM.knex
+      .select(columnsToCopy)
+      .from(model.collectionName)
+      .whereRaw('?', 0),
+  ]);
+
+  // Transaction is started after DDL because of MySQL (https://dev.mysql.com/doc/refman/8.0/en/implicit-commit.html)
+  const trx = await ORM.knex.transaction();
+
+  // bulk insert updates in tmp table
+  try {
+    const processedLocaleCodes = [];
+    for (const locale of locales) {
+      const batchSize = BATCH_SIZE;
+      let offset = 0;
+      let batchCount = BATCH_SIZE;
+      while (batchCount === batchSize) {
+        const batch = await trx
+          .select([...attributesToMigrate, 'locale', 'localizations'])
+          .from(model.collectionName)
+          .where('locale', locale.code)
+          .orderBy('id')
+          .offset(offset)
+          .limit(batchSize);
+
+        // postgres automatically parses JSON, but not slite nor mysql
+        batch.forEach(entry => {
+          if (typeof entry.localizations === 'string') {
+            entry.localizations = JSON.parse(entry.localizations);
+          }
+        });
+
+        batchCount = batch.length;
+        const entriesToProcess = batch.filter(shouldBeProceed(processedLocaleCodes));
+
+        const tempEntries = entriesToProcess.reduce((entries, entry) => {
+          const attributesValues = pick(attributesToMigrate, entry);
+          const entriesIdsToUpdate = entry.localizations
+            .filter(related => related.locale !== locale.code)
+            .map(prop('id'));
+
+          return entries.concat(entriesIdsToUpdate.map(id => ({ id, ...attributesValues })));
+        }, []);
```

Comment:
You can pull some logic outisde the reduce to reuse between bookshelf & mongoose